File: ALifeCommon/src/main/java/com/cristiano/alife/instructions/transfers/LoadFromRegisterInstruction.java
package com.cristiano.alife.instructions.transfers;

import com.cristiano.alife.consts.ALifeConsts;
import com.cristiano.alife.world.BaseInstruction;
import com.cristiano.alife.world.IOrganismo;
import com.cristiano.alife.world.IWorld;

/*
 * Load a byte from the "from" register position to the "to" register: 
 * LOADI: to <- (from)
 * */
public class LoadFromRegisterInstruction extends BaseInstruction  {
	public LoadFromRegisterInstruction(IWorld mundo) {
		super(mundo);
	}
	@Override
	public void executa(IOrganismo o) {
		int fromPos = o.getReg(getByteOrganismo(o,2));
		int inst=o.getMemory(fromPos);
		inst=mundo.getMutation().mutateInstruction(inst,o);
		/*if (!mundo.isValidInstruction(inst)){
			o.fatalError();
			return;
		}*/
		o.setReg(getByteOrganismo(o,1),inst);
	}
	
	@Override
	public String getDescription(IOrganismo o,int ip){
		return comment(name+" "+ALifeConsts.getLetter(o.getMemory(ip+1))+","+ALifeConsts.getLetter(o.getMemory(ip+2))
				,ALifeConsts.getLetter(o.getMemory(ip+1))+" <- ("+ALifeConsts.getLetter(o.getMemory(ip+2))+")");
	}
	
	@Override
	public int getStep() {
		return 3;
	}
	

}

File: netAvida/netAvida.Backend/instructions/impl/transfers/LoadFromRegisterInstruction.cs
using netAvida.backend;
using netAvida.backend.interfaces;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace netAvida.Backend.instructions.impl
{
    public class LoadFromRegisterInstruction : BaseInstruction
    {
        public LoadFromRegisterInstruction(IWorld mundo): base(mundo)
        {
        }

    public override void executa(IOrganismo o)
        {
            int fromPos = o.getReg(getByteOrganismo(o, 2));
            int inst = o.getMemory(fromPos);
            inst = mundo.getMutation().mutateInstruction(inst, o);
            /*if (!mundo.isValidInstruction(inst)){
                o.fatalError();
                return;
            }*/
            o.setReg(getByteOrganismo(o, 1), inst);
        }

    public override int getStep()
        {
            return 3;
        }

    public override String getDescription(IOrganismo o, int ip)
        {
            return comment(name + " " + ALifeConsts.getLetter(o.getMemory(ip + 1)) + "," + ALifeConsts.getLetter(o.getMemory(ip + 2))
                  , ALifeConsts.getLetter(o.getMemory(ip + 1)) + " <- (" + ALifeConsts.getLetter(o.getMemory(ip + 2)) + ")");
        }
    }
}
